src/storage/secondary/compactor.rs

```diff
@@ -97,10 +100,44 @@ impl Compactor {
             ConcatIterator::new(iters).into()
         };
 
-        let mut builder = RowsetBuilder::new(
-            table.columns.clone(),
-            ColumnBuilderOptions::from_storage_options(&table.storage_options),
-        );
+        let mut distinct_value = 0;
+        let mut row_count = 0;
+
+        {
+            let mut distinct_value_aggregator =
+                create_statistics_global_aggregator(BlockStatisticsType::DistinctValue);
+            let mut row_count_aggregator =
+                create_statistics_global_aggregator(BlockStatisticsType::RowCount);
+
+            for rowset in &selected_rowsets {
+                let columns = rowset.get_columns();
+                let column_indexs = columns.iter().map(|column| column.index()).collect_vec();
+                for column_index in column_indexs {
+                    row_count_aggregator.apply_batch(column_index);
+                    distinct_value_aggregator.apply_batch(column_index);
+                }
+            }
+
+            if let DataValue::Int64(dv) = distinct_value_aggregator.get_output() {
+                distinct_value = dv;
+            }
+
+            if let DataValue::Int64(rc) = row_count_aggregator.get_output() {
+                row_count = rc;
+            }
+        }
+
+        let mut builder = if distinct_value < row_count / 5 {
+            RowsetBuilder::new(
+                table.columns.clone(),
+                ColumnBuilderOptions::from_storage_options(&table.storage_options),
+            )
+        } else {
+            let mut column_options =
+                ColumnBuilderOptions::from_storage_options(&table.storage_options);
+            column_options.encode_type = EncodeType::RunLength;
```

Comment:
We can use `DictEncoding`. If there're few distinct values but the data are like `1,2,3,1,2,3,1,2,3,...`, RLE might perform badly.

Reply:
Dict encoding for f64 is not supported yet. I couldn't pass risinglight_sqllogictest::sqllogictest disk::limit.slt

Reply:
Makes sense... https://github.com/risinglightdb/risinglight/issues/670

Then let's try RLE first.